escrow/lib/go/test/escrow_test.go

```diff
@@ -0,0 +1,691 @@
+		)
+	})
+
+	t.Run("Should not be able to mint an edition that has reached max minting size", func(t *testing.T) {
+		testMintMomentNFT(
+			t,
+			b,
+			contracts,
+			uint64(1),
+			nil,
+			userAddress,
+			uint64(3),
+			uint64(3),
+			true,
+		)
+	})
+
+	t.Run("Should not be able to mint an edition that is already closed", func(t *testing.T) {
+		testMintMomentNFT(
+			t,
+			b,
+			contracts,
+			uint64(3),
+			nil,
+			userAddress,
+			uint64(1),
+			uint64(1),
+			true,
+		)
+	})
+}
+
+func testMintMomentNFT(
+	t *testing.T,
+	b *emulator.Blockchain,
+	contracts Contracts,
+	editionID uint64,
+	serialNumber *uint64,
+	userAddress flow.Address,
+	shouldBeID uint64,
+	shouldBeSerialNumber uint64,
+	shouldRevert bool,
+) {
+	// Make sure the total supply of NFTs is tracked correctly
+	previousSupply := getMomentNFTSupply(t, b, contracts)
+
+	mintMomentNFT(
+		t,
+		b,
+		contracts,
+		userAddress,
+		editionID,
+		serialNumber,
+		shouldRevert,
+	)
+
+	newSupply := getMomentNFTSupply(t, b, contracts)
+	if !shouldRevert {
+		assert.Equal(t, previousSupply+uint64(1), newSupply)
+
+		nftProperties := getMomentNFTProperties(
+			t,
+			b,
+			contracts,
+			userAddress,
+			shouldBeID,
+		)
+		assert.Equal(t, shouldBeID, nftProperties.ID)
+		assert.Equal(t, editionID, nftProperties.EditionID)
+		assert.Equal(t, shouldBeSerialNumber, nftProperties.SerialNumber)
+		assert.Equal(t, shouldBeSerialNumber, nftProperties.SerialNumber)
+		assert.Less(t, uint64(0), nftProperties.MintingDate)
+	} else {
+		assert.Equal(t, previousSupply, newSupply)
+	}
+}
+
+// ------------------------------------------------------------
+// Escrow
+// ------------------------------------------------------------
+func TestEscrow(t *testing.T) {
+	b := newEmulator()
+	contracts := EscrowContracts(t, b)
+	userAddress, userSigner := createAccount(t, b)
+	setupAllDay(t, b, userAddress, userSigner, contracts)
+
+	createTestEditions(t, b, contracts)
+
+	t.Run("Should be able to mint a new MomentNFT from an edition that has a maxMintSize", func(t *testing.T) {
+		testMintMomentNFT(
+			t,
+			b,
+			contracts,
+			uint64(1),
+			nil,
+			userAddress,
+			uint64(1),
+			uint64(1),
+			false,
+		)
+	})
+
+	t.Run("Should confirm that 1 MomentNFT exists within users collection", func(t *testing.T) {
+		// Get the MomentNFT data from the users collection.
+		count := getMomentNFTLengthInAccount(t, b, contracts, userAddress)
+		assert.Equal(t, big.NewInt(1), count)
+	})
+
+	t.Run("Should be able to create a leaderboard", func(t *testing.T) {
+		testCreateLeaderboard(
+			t,
+			b,
+			contracts,
+			"leaderboardBurn-1",
+		)
+	})
+
+	t.Run("Should get the leaderboard by name to confirm it exists", func(t *testing.T) {
+		// Get leaderboard data from the contract.
+		leaderboard, _ := getLeaderboardData(t, b, contracts, "leaderboardBurn-1")
+		assert.Equal(t, "\"leaderboardBurn-1\"", leaderboard.Name)
+		assert.Equal(t, "Type<A.e03daebed8ca0615.AllDay.NFT>()", leaderboard.NftType)
```

Comment:
Is it safe to hardcode this address? (I suppose the address would be deterministic, but might be brittle if accounts are created in another order?)

Reply:
Yea! So this is the NFTType of the Leaderboard we create during the transaction. No other NFTs other than this type are allowed inside. Does that make sense?

Reply:
A bit confused! Wouldn't the address that the AllDay contract gets deployed to (and thus the type) be defendant on the order that accounts are are created?